zap/src/main/java/org/parosproxy/paros/model/Session.java

```diff
@@ -1636,6 +1648,135 @@ public ParameterParser getFormParamParser(String url) {
+            params.addAll(
+                    convertNVP(
+                            model.getSession().getParameters(msg, Type.form),
+                            org.parosproxy.paros.core.scanner.NameValuePair.TYPE_POST_DATA));
+        }
+
+        return getLeafName(
+                nodeName,
+                msg.getRequestHeader().getMethod(),
+                msg.getRequestHeader().getHeader(HttpHeader.CONTENT_TYPE),
+                params);
+    }
+
+    /**
+     * Gets the name of the node to be used for the given parameters in the Site Map.
+     *
+     * @param nodeName the last element of the path
+     * @param uri
+     * @param method
+     * @param postData
+     * @return the name of the node to be used in the Site Map
+     * @throws HttpMalformedHeaderException
+     * @since TODO add version
+     */
+    public String getLeafName(String nodeName, URI uri, String method, String postData)
+            throws HttpMalformedHeaderException {
+        HttpMessage msg = new HttpMessage(uri);
+        msg.getRequestHeader().setMethod(method);
+        if (method.equalsIgnoreCase(HttpRequestHeader.POST)) {
+            msg.getRequestBody().setBody(postData);
+            msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+        }
+        return getLeafName(nodeName, msg);
+    }
+
+    private List<org.parosproxy.paros.core.scanner.NameValuePair> convertNVP(
+            List<NameValuePair> nvpList, int type) {
+        List<org.parosproxy.paros.core.scanner.NameValuePair> params =
+                new ArrayList<org.parosproxy.paros.core.scanner.NameValuePair>();
+        for (NameValuePair nvp : nvpList) {
+            params.add(
+                    new org.parosproxy.paros.core.scanner.NameValuePair(
+                            type, nvp.getName(), nvp.getValue(), -1));
+        }
+        return params;
+    }
+
+    /**
+     * Returns the names of the parameters in a form suitable to display in the Site Map
+     *
+     * @param list the full set of parameters
+     * @param type the type of parameters to be included
+     * @return the names of the given parameters
+     * @since TODO add version
+     */
+    private static String getParamDisplayString(
+            List<org.parosproxy.paros.core.scanner.NameValuePair> list, int type) {
+        StringBuilder sb = new StringBuilder();
+        list.stream()
+                .sorted()
+                .filter(entry -> entry.getType() == type)
+                .forEach(
+                        entry -> {
+                            String name = entry.getName();
+                            if (name != null) {
+                                if (sb.length() > 0) {
+                                    sb.append(',');
+                                }
+                                if (name.length() > 40) {
+                                    // Truncate
+                                    name = name.substring(0, 40);
```

Comment:
I think we have a utility function for doing this that also inserts an ellipsis `...` so that it's more obvious that it was intentionally truncated?